src/toil/batchSystems/mesos/executor.py

```diff
@@ -212,11 +233,75 @@ def frameworkMessage(self, driver, message):
         log.debug("Received message from framework: {}".format(message))
 
 
-def main(executorClass=MesosExecutor):
+def main():
     logging.basicConfig(level=logging.DEBUG)
     log.debug("Starting executor")
-    executor = executorClass()
-    driver = mesos.native.MesosExecutorDriver(executor)
-    exit_value = 0 if driver.run() == mesos_pb2.DRIVER_STOPPED else 1
+
+    if not os.environ.has_key("MESOS_AGENT_ENDPOINT"):
+        # Some Mesos setups in our tests somehow lack this variable. Provide a
+        # fake one to maybe convince the executor driver to work.
+        os.environ["MESOS_AGENT_ENDPOINT"] = os.environ.get("MESOS_SLAVE_ENDPOINT", "127.0.0.1:5051")
+        log.warning("Had to fake MESOS_AGENT_ENDPOINT as %s" % os.environ["MESOS_AGENT_ENDPOINT"])
+        
+    try:
+        urlopen("http://%s/logging/toggle?level=1&duration=15mins" % os.environ["MESOS_AGENT_ENDPOINT"]).read()
```

Comment:
We might want to ditch this log level toggle. I had it in there to try and make debugging failing executors easier, but I don't think I ever got much useful out of it, and we probably don't want to be tinkering with people's settings in production.

Reply:
I put a manual switch in.  Not sure if it's the best method, but if it ever comes up again someone reading the code might find it useful.